dt-demo/dimensions/src/test/java/com/datatorrent/demos/dimensions/telecom/DataWrapper.java

```diff
@@ -0,0 +1,139 @@
+package com.datatorrent.demos.dimensions.telecom;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+/**
+ * keep all instances generated at runtime
+ * 
+ * There could different type of instances of this class.
+ * The instances which was just cloned should have the same id.
+ * The really new instances should have a different id from other.
+ * 
+ * LIMITATION: this class can only used for one JVM, typically for unit test
```

Comment:
I hope there are simple solution which can avoid the one JVM limitation. But it would be something like zookeeper.